src/utils/FlowUtils.ts

```diff
@@ -1,28 +1,59 @@
 import { Deposit, Fill } from "../interfaces";
+import { isV2Deposit, isV3Deposit, isV2Fill, isV3Fill } from "./V3Utils";
 
 export const FILL_DEPOSIT_COMPARISON_KEYS = [
-  "amount",
-  "originChainId",
-  "relayerFeePct",
-  "realizedLpFeePct",
   "depositId",
+  "originChainId",
+  "destinationChainId",
   "depositor",
   "recipient",
-  "destinationChainId",
-  "destinationToken",
   "message",
 ] as const;
 
+export const V2_DEPOSIT_COMPARISON_KEYS = [
+  ...FILL_DEPOSIT_COMPARISON_KEYS,
+  "amount",
+  "destinationToken",
+  "relayerFeePct",
+  "realizedLpFeePct",
+] as const;
+
+export const V3_DEPOSIT_COMPARISON_KEYS = [
+  ...FILL_DEPOSIT_COMPARISON_KEYS,
+  "inputToken",
+  "inputAmount",
+  "outputToken",
+  "outputAmount",
+] as const;
+
 export function filledSameDeposit(fillA: Fill, fillB: Fill): boolean {
-  return (
-    fillA.depositId === fillB.depositId &&
-    fillA.originChainId === fillB.originChainId &&
-    fillA.amount.eq(fillB.amount) &&
-    fillA.destinationChainId === fillB.destinationChainId &&
-    fillA.relayerFeePct.eq(fillB.relayerFeePct) &&
-    fillA.recipient === fillB.recipient &&
-    fillA.depositor === fillB.depositor
-  );
+  if (isV2Fill(fillA) && isV2Fill(fillB)) {
+    return (
+      fillA.depositId === fillB.depositId &&
+      fillA.originChainId === fillB.originChainId &&
+      fillA.amount.eq(fillB.amount) &&
+      fillA.destinationChainId === fillB.destinationChainId &&
+      fillA.relayerFeePct.eq(fillB.relayerFeePct) &&
+      fillA.recipient === fillB.recipient &&
+      fillA.depositor === fillB.depositor &&
+      fillA.message === fillB.message
+    );
+  } else if (isV3Fill(fillA) && isV3Fill(fillB)) {
+    return (
+      fillA.depositId === fillB.depositId &&
```

Comment:
We also need to compare `exclusiveRelayer`, `fillDeadline`, `exclusivityDeadline`. Basically everything in `V3RelayData`. So should we use `getV3RelayHash` to compare?

Reply:
That's a good proposal and would make life easier when comparing deposits or fills. I'll implement that.